src/util/format_expr.h

```diff
@@ -10,11 +10,18 @@ Author: Daniel Kroening, kroening@kroening.com
 #define CPROVER_UTIL_FORMAT_EXPR_H
 
 #include "format.h"
+#include "irep.h"
+
+#include <functional>
 
 class exprt;
 
 //! Formats an expression in a generic syntax
 //! that is inspired by C/C++/Java, and is meant for debugging
 std::ostream &format_rec(std::ostream &, const exprt &);
 
+void add_format_hook(
+  irep_idt,
+  std::function<std::ostream &(std::ostream &, const exprt &)>);
```

Comment:
Needs some documentation (to explain what the parameters are and what the expected behaviour of that function is). A (unit) test would also be nice.

Reply:
Both added